vmngclient/api/task_status_api.py

```diff
@@ -106,7 +114,18 @@ def wait_for_action_finish() -> TaskStatus:
             TaskStatus: TaskStatus instance
         """
         url = f"{action_url}{action_id}"
-        action_data = session.get_data(url)[0]
+        try:
+            action_data = session.get_data(url)[0]
+        except IndexError:
+            tasks_ids = get_all_tasks()
+            if action_id in tasks_ids:
+                sleep(delay_seconds)
+                try:
+                    action_data = session.get_data(url)[0]
+                except IndexError:
+                    raise ValueError(f"task id {action_id} is not registered by vManage.")
```

Comment:
This ValueError is already in the `if action_id in tasks_ids:` condition, so task was registered (it is on the list) but there is another problem? Can that happen?

Reply:
We decided in point 2.2.3, that if there no proper response after sleep, error is being raised. Probably, if task is registered, this situation is not possible. But it's just "probably".

Reply:
What about raising in this particular line something like:
```suggestion
                    raise IndexError(f"task id {action_id} registered by vManage in all tasks list, but response about its status didn't contain any information.")
```
? Just to inform user that this case happened and have different Errors raised